lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
+    # Translate
+    #   VALUE args[6], failed, a1, a2, a3, a4, a5, a6;
+    #  into
+    #   VALUE failed, a1, a2, a3, a4, a5, a6; VALUE *args = truffle_managed_malloc(6 * sizeof(VALUE));
+
+    simple = []
+    arrays = []
+
+    line = $1
+
+    $2.split(', ').each do |local|
```

Comment:
Also, the space after `,` is not necessary in C (`VALUE a,b,c;`).
Makes me think it would be great to have simple tests for this preprocessor:
```ruby
assert_equal expected, preprocess(input)
```